build/xtask/src/dist.rs

```diff
@@ -2052,15 +2060,15 @@ fn get_git_status() -> Result<(String, bool)> {
 
 fn binary_to_srec(
     binary: &Path,
-    bin_addr: u32,
-    entry: u32,
+    bin_addr: AbiSize,
+    entry: AbiSize,
     out: &Path,
 ) -> Result<()> {
     let mut srec_out = vec![srec::Record::S0("signed".to_string())];
 
     let binary = std::fs::read(binary)?;
 
-    let mut addr = bin_addr;
+    let mut addr = bin_addr as u32;
```

Comment:
Is there anything to check that the `bin_addr` will fit in a `u32` before this? If not, an error should be generated instead of silently ignoring that.

Reply:
Fixed in next patchset.